### PR TITLE
Springboot upgrade version 2.7.x

### PR DESCRIPTION
**Pull request by rkumar8j**
#102 
sol005 driver Spring boot version Upgrade to 2.6.x/2.7.x 
Signed-off-by: rkumar8j [rkumar8j@in.ibm.com](mailto:rkumar8j@in.ibm.com)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -40,9 +40,11 @@
         <helm.repo.api.url>${helm.repo.addr}/api/accanto</helm.repo.api.url>
         <helm.repo.snapshots.url>${helm.repo.addr}/accanto/snapshots</helm.repo.snapshots.url>
         <helm.repo.url>${helm.repo.addr}/accanto/stable</helm.repo.url>
+        <httpclient.version>4.5.13</httpclient.version>
         <jacoco-maven-plugin.version>0.8.4</jacoco-maven-plugin.version>
         <java.version>11</java.version>
-        <kiwigrid.version>4.13</kiwigrid.version>
+        <kiwigrid.version>2.5</kiwigrid.version>
+        <log4j.version>2.17.1</log4j.version>
```
Comment (lokanalla) [inline pom.xml:47]: You can remove log4j.

File: pom.xml [hunk 2]
```diff
@@ -56,9 +58,9 @@
         <sortpom-maven-plugin.version>2.8.0</sortpom-maven-plugin.version>
         <spring-beans.version>5.3.21</spring-beans.version>
```
Comment (lokanalla) [inline pom.xml:59]: Try to remove explicit spring beans and core dependencies and see build succeeds.

File: pom.xml [hunk 3]
```diff
@@ -79,58 +81,39 @@
...
-            <groupId>io.swagger.core.v3</groupId>
-            <artifactId>swagger-annotations</artifactId>
-            <version>2.1.11</version>
-        </dependency>
-        <dependency>
-            <groupId>javax.validation</groupId>
-            <artifactId>validation-api</artifactId>
-            <version>2.0.1.Final</version>
-        </dependency>
         <dependency>
```
Comment (lokanalla) [inline pom.xml:85]: Check whether encoder can also be removed.

File: pom.xml [hunk 4]
```diff
@@ -79,58 +81,39 @@
...
-            <version>2.4.8</version>
+            <groupId>org.apache.httpcomponents</groupId>
+            <artifactId>httpclient</artifactId>
+            <version>${httpclient.version}</version>
         </dependency>
         <dependency>
-            <groupId>org.apache.commons</groupId>
-            <artifactId>commons-lang3</artifactId>
-            <version>3.12.0</version>
+            <groupId>org.apache.logging.log4j</groupId>
```
Comment (lokanalla) [inline pom.xml:96]: remove log4j dependency

File: pom.xml [hunk 5]
```diff
@@ -202,7 +182,7 @@
         <dependency>
```
Comment (lokanalla) [inline pom.xml:182]: Check whether you can remove oauth2 dependency as well.

**Review by lokanalla — CHANGES_REQUESTED** [review]
Please remove explicit dependencies and try to build and test.

File: pom.xml (hunk 5 above)
Comment (rkumar8j) [reply pom.xml:182]: No, Build is failing because of the direct dependency in one of the class.

File: pom.xml (hunk 1 above)
Comment (rkumar8j) [reply pom.xml]: Done

File: pom.xml (hunk 2 above)
Comment (rkumar8j) [reply pom.xml]: Done

File: pom.xml (hunk 3 above)
Comment (rkumar8j) [reply pom.xml:80]: Done

File: pom.xml (hunk 4 above)
Comment (rkumar8j) [reply pom.xml]: Done

**Review by lokanalla — APPROVED** [review]

**Pull request merged**